code/game/machinery/computer/robot.dm

```diff
@@ -88,21 +91,38 @@
 			if(allowed(usr))
 				var/mob/living/silicon/robot/R = locate(params["ref"]) in GLOB.silicon_mobs
 				if(can_control(usr, R) && !..())
-					if(isAI(usr) && (R.ai_lockdown && R.lockcharge || !R.lockcharge) || !isAI(usr))
-						R.ai_lockdown = FALSE
-						if(isAI(usr) && !R.lockcharge)
+					if(isAI(usr))
+						if(R.ai_lockdown && R.lockcharge)
+							R.ai_lockdown = FALSE
+							src.lock_unlock_borg(R)
+						else if(!R.ai_lockdown && R.lockcharge)
+							to_chat(usr, span_danger("Cyborg locked by an user with superior permissions."))
+						else if(!R.lockcharge)
```

Comment:
Can we reformat to something like
```
if (isAI(user))
   if (R.lockcharge)
      if(R.ai_lockdown)
         release ai lockdown
      else
         show error message
   else
      apply ai lockdown
```
This would still be a horrible mess of branching cases but I think that might be a bit easier to follow? And saves on repeating the same check.